watttime/tcy.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import pandas as pd
+import pytz
+from typing import Optional, List
+from .api import WattTimeHistorical
+
+@dataclass
+class TCYConfig:
+    """Configuration for TCY calculation"""
+    region: str
+    timezone: str
+    holidays: Optional[List[str]] = None
+
+class TCYCalculator:
+    """Calculate Typical Carbon Year profiles from historical MOER data using a 3-year lookback period"""
+    
+    def __init__(self, username: str, password: str, config: TCYConfig):
```

Comment:
```suggestion
    def __init__(self, username: Optional[str] = None, password: Optional[str] = None, config: TCYConfig):
```

This lets you skip specifying username and password, which will use the env vars when `None`s are passed to `WattTimeHistorical`